src/api/alchemy.ts

```diff
@@ -525,9 +929,17 @@ export class Alchemy {
   }
 
   /**
-   * Creates an AlchemyWebsocketProvider instance. Only one provider is created
+   * Returns an AlchemyWebsocketProvider instance. Only one provider is created
    * per Alchemy instance.
    *
+   * The AlchemyWebSocketProvider is a wrapper around ether's
+   * `AlchemyWebSocketProvider` class and has been expanded to support Alchemy's
+   * Subscription APIs, automatic backfilling, and other performance improvements.
+   *
+   * Most common methods on the provider are available as top-level methods on
+   * the {@link Alchemy} instance, but the provider is exposed here to access
+   * other less-common methods.
+   *
    * @public
    */
   async getWebsocketProvider(): Promise<AlchemyWebSocketProvider> {
```

Comment:
Yo, I just noticed: isn't this subject to race conditions? If someone were to call this multiple times before the import completes, we'd end up creating multiple websocket providers, which is an issue because we'll end up with multiple connections.

Seems like we want something like
```ts
private wssProviderPromise: Promise<AlchemyWebSocketProvider> | undefined;

getWebsocketProvider(): Promise<AlchemyWebSocketProvider> {
  if (!this.wssProviderPromise) {
    this.wssProviderPromise = (async () => {
      const { AlchemyWebSocketProvider } = await import(
        './alchemy-websocket-provider'
      );
      return new AlchemyWebSocketProvider(
        this.network,
        this.apiKey
      );
    })();
  }
  return this.wssProviderPromise;
}

}
```

Reply:
wow, that's a good catch, thanks! I guess the average person won't be calling `getProvider()` without an `await` in front of it, but that's still a bad bug.

Added a test that verifies we don't race.

Reply:
Yea, you're right most people likely won't hit it. It's a bit more likely now that we have top-level methods, so someone could be like
```ts
const alchemy = new Alchemy(stuff);
alchemy.on("blocks", () => { /* ... */ });
alchemy.on("logs", () => { /* ... */ });
```